Tiltfile

```diff
@@ -395,16 +532,38 @@ def prepare_all():
                 context = context,
             )
 
+    addons_arg = ""
+    for name in settings.get("deploy_addons", []):
+        addon = addons.get(name)
+        if addon == None:
+            fail("Addon with name " + name + " not found")
+        if addon.get("kustomize_config", True):
+            addon_substitutions = addon.get("kustomize_substitutions", {})
+            for addon_substitution in addon_substitutions:
+                if addon_substitution in kustomize_substitutions:
+                    if kustomize_substitutions[addon_substitution] != addon_substitutions[addon_substitution]:
+                        fail("Duplicate kustomize substitution key with different values for " + addon_substitution)
+                kustomize_substitutions[addon_substitution] = addon_substitutions[addon_substitution]
+            context = addon.get("context")
+            addons_arg = addons_arg + "--addons {name}:{container_name}:{binary_name}:{context} ".format(
+                name = name,
+                container_name = addon.get("container_name"),
+                binary_name = addon.get("binary_name"),
+                context = context,
+            )
     tilt_settings_file_arg = "--tilt-settings-file " + tilt_file
 
-    cmd = "make -B tilt-prepare && ./hack/tools/bin/tilt-prepare {allow_k8s_arg}{tools_arg}{cert_manager_arg}{kustomize_build_arg}{providers_arg}{tilt_settings_file_arg}".format(
+    cmd = "make -B tilt-prepare && ./hack/tools/bin/tilt-prepare {allow_k8s_arg}{tools_arg}{cert_manager_arg}{kustomize_build_arg}{providers_arg}{addons_arg}{tilt_settings_file_arg}".format(
         allow_k8s_arg = allow_k8s_arg,
         tools_arg = tools_arg,
         cert_manager_arg = cert_manager_arg,
         kustomize_build_arg = kustomize_build_arg,
         providers_arg = providers_arg,
+        addons_arg = addons_arg,
         tilt_settings_file_arg = tilt_settings_file_arg,
     )
+
+    env = kustomize_substitutions
```

Comment:
Out of curiosity: why do we need this? Won't the `env` argument to `local` command in l.567 take care of setting the environment variables appropriately when running the command?

Reply:
Good catch!